vazhin/app.js

```diff
@@ -0,0 +1,283 @@
+// Selectors
+
+const todayDate = document.querySelector(".today-date");
+const todoList = document.querySelector(".todo-list");
+const taskText = document.querySelector('#todo-text');
+const priorityInput = document.querySelector('#select-priority');
+const deadlineInput = document.querySelector('#date');
+const addBtn = document.querySelector(".btn");
+const cancelBtn = document.querySelector(".cancel-btn");
+const noTasks = document.querySelector(".tasks-empty");
+let arrOfTasks = [];
+let inEditMode = false;
+let indexOfInEditTask;
+priorityInput.value = "";
+
+// Event listeners
+
+window.addEventListener('load', startDateTime);
+addBtn.addEventListener("click", addTodo);
+cancelBtn.addEventListener("click", cancelTodo);
+
+// Functions
+
+function addTodo(event) {
+    event.preventDefault();
+
+    if (taskText.value === "") {
+
+        taskText.classList.add("fill-out-field");
+        taskText.placeholder = "add a task";
+        return;
+    }
+    else if (priorityInput.value === "") {
+
+        taskText.classList.remove("fill-out-field");
+        priorityInput.classList.add("fill-out-field");
+        priorityInput.placeholder = "select priority";
+        return;
+    }
+    else if (deadlineInput.value === "") {
+
+        priorityInput.classList.remove("fill-out-field");
+        deadlineInput.classList.add("fill-out-field");
+        deadlineInput.placeholder = "enter a deadline";
+        return;
+    }
+    else {
+        taskText.placeholder = "coding... 👩‍💻";
+        priorityInput.placeholder = '';
+        deadlineInput.placeholder = '';
+        deadlineInput.classList.remove("fill-out-field");
+
+        const objOfTask = {
+            task: taskText.value,
+            priority: priorityInput.value,
+            deadline: deadlineInput.value,
+        };
+
+        if (inEditMode) {
+            arrOfTasks.splice(indexOfInEditTask, 1, objOfTask);
+            inEditMode = false;
+            addBtn.innerHTML = "add task"
+        }
+        else {
+            arrOfTasks.push(objOfTask);
+        }
+    }
+
+    updateTasksInList();
+}
+
+function updateTasksInList() {
+
+    let previousTasks = document.querySelectorAll(".todo-list div");
+    for (let i = 0; i < previousTasks.length; i++) {
```

Comment:
instead of the for-loop, you could do the following
`let tasksList = document.querySelector(".todo-list");`
`tasksList.innerHTML = ""`
it would empty everything.